student/src/main/java/tds/student/web/handlers/TestResponseHandler.java

```diff
@@ -128,6 +137,17 @@ private void updateResponses (HttpServletRequest request, HttpServletResponse re
       // get the request information from tehe browser
       TestResponseReader responseReader = TestResponseReaderSax.parseSax (request.getInputStream (), testOpp);
 
+      List<Map<String,String>> responses = new ArrayList<>();
+      for(ItemResponseUpdate responseUpdate: responseReader.getResponses()) {
```

Comment:
I mentioned this is in @mtrupkin 's review, but I wonder if there is any value in applications being able to register `ObjectLogEventHandlers` that could handle these types of event logging.  So the block from 140 - 149 would be in one of the handlers and this would be reduced to `_eventLogger.info(responseReader)` with the event logger knowing how to get the data it wants rather than sprinkling that logic into the controllers